bulla-contracts/src/functions/common.ts

```diff
@@ -359,6 +376,12 @@ function mulDiv(x: BigInt, y: BigInt, denominator: BigInt, roundingUp: boolean =
   return result;
 }
 
+export const getTrueProtocolFeeV3_1 = (poolAddress: Address, invoiceId: BigInt): BigInt => {
```

Comment:
can't we get the invoicefunded event and get the targetProtocolFee instead? would avoid an rpc call.